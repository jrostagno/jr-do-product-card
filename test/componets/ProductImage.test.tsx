import React from 'react';

import { render } from '@testing-library/react';

import { ProductCard } from '../../src/components/ProductCard';
import { ProductImage } from '../../src/components/ProductImage';
import { product2 } from '../data/product';

describe('ProductImage', () => {
  test('Debe mostrar el componente con la Imagen  personalizado', () => {
    const { asFragment } = render(
      <ProductImage img="https:://hola.jpg" className="custom-title" />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('Debe de mostrar el componente con la imagen del producto', () => {
    const product = render(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );

    expect(product.rerender).toMatchSnapshot();
  });
});
