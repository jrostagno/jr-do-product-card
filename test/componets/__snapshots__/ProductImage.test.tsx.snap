// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Debe de mostrar el componente con la imagen del producto 1`] = `[Function]`;

exports[`ProductImage Debe mostrar el componente con la Imagen  personalizado 1`] = `
<DocumentFragment>
  <img
    alt="Cofee mug"
    class="productImg custom-title "
    src="https:://hola.jpg"
  />
</DocumentFragment>
`;
